java/metrics/src/main/scala/org/typelevel/otel4s/java/metrics/ObservableCounterBuilderImpl.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+package java
+package metrics
+
+import cats.effect.kernel.Async
+import cats.effect.kernel.Resource
+import cats.effect.std.Dispatcher
+import cats.syntax.all._
+import io.opentelemetry.api.metrics.{Meter => JMeter}
+import org.typelevel.otel4s.metrics._
+
+private[java] case class ObservableCounterBuilderImpl[F[_]](
+    jMeter: JMeter,
+    name: String,
+    unit: Option[String] = None,
+    description: Option[String] = None
+)(implicit F: Async[F])
+    extends ObservableInstrumentBuilder[F, Long, ObservableCounter] {
+
+  type Self = ObservableCounterBuilderImpl[F]
+
+  def withUnit(unit: String): Self = copy(unit = Option(unit))
+  def withDescription(description: String): Self =
+    copy(description = Option(description))
+
+  def createWithCallback(
+      cb: ObservableMeasurement[F, Long] => F[Unit]
+  ): Resource[F, ObservableCounter] =
+    Dispatcher.parallel.flatMap(dispatcher =>
```

Comment:
I'm wondering which dispatcher is more suitable. The initiator of the callback evaluation is OpenTelemtry internals (it's a `MetricProducer#collectAllMetrics`, I believe). 

Should we guarantee the order?

Reply:
I'm definitely not an expert on `Dispatcher` semantics. I'll just change this to sequential, since these callbacks are called quite rarely, so I don't think performance should be a worry.

Reply:
I don't think it's going to matter a whole lot.  My gut feeling is sequential is right.